src/main.c

```diff
@@ -1,27 +1,99 @@
+/***********************************************************************
+ *
+ * Copyright (c) 2021
+ * IoTech Ltd
+ *
+ **********************************************************************/
+
 #include <stdio.h>
+#include <unistd.h>
 
-#include <dbus/dbus.h>
+#include "dbusutils.h"
+#include "device.h"
+#include "service.h"
+#include "characteristic.h"
+#include "descriptor.h"
 
-int main (int argc, char *argv[]) 
+DBusConnection* global_dbus_connection;
+
+static DBusHandlerResult filter_message (DBusConnection *connection, DBusMessage *message, void *data)
+{
+
+  printf("Incomming DBus Message %d %s : %s %s/%s/%s %s\n",
+          dbus_message_get_type (message),
+          dbus_message_get_sender (message),
+          dbus_message_get_destination (message),
+          dbus_message_get_path (message),
+          dbus_message_get_interface (message),
+          dbus_message_get_member (message),
+          dbus_message_get_type (message) == DBUS_MESSAGE_TYPE_ERROR ?
+          dbus_message_get_error_name (message) : ""
+        );
+
+  // if (dbus_message_is_method_call(msg, "test.method.Type", "Method"))
+  // if (dbus_message_is_signal(msg, "test.signal.Type", "Test")) 
```

Comment:
Question: Why are these commented out? If its going to be used later could add in a TODO